demo.ipynb

```diff
@@ -3386,7 +3386,7 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.9.12"
```

Comment:
Does this also run for 3.9?

Reply:
The codebase was run on 3.10.9, and since version-specific functions were not used, there shouldn't be any problem with running it on 3.9.